### ISSUE TITLE
Nested selector should not be case-sensitive 

### ISSUE THREAD

**Issue by ocavue** [jsdom/jsdom#3544]
### Basic info:

- **Node.js version:** 18
- **jsdom version:** 21.1.2

After updating JSDOM from 21.1.1 to 21.1.2 (and nwsapi from 2.2.2 to 2.2.4 https://github.com/jsdom/jsdom/commit/f75a9d1f7c27d653800b128ac75e2d9f0e93cb1f), the nested `UL > LI` doesn't work anymore because it becomes case-sensitive .

### Minimal reproduction case

Given the following code, it should print `1 1 1 1`. However it actually prints `1 1 1 0` 

```ts
import { JSDOM } from 'jsdom';

const dom = new JSDOM(`<!DOCTYPE html><body></body>`);
const document = dom.window.document;
const body = document.body;

const div = document.createElement('div');
const ul = document.createElement('ul');
const li = document.createElement('li');

ul.appendChild(li);
div.appendChild(ul);
body.appendChild(div);

console.log(div.querySelectorAll('UL').length);
console.log(div.querySelectorAll('ul').length);
console.log(div.querySelectorAll('ul > li').length);
console.log(div.querySelectorAll('UL > LI').length);
```

Online playground with JSDOM 21.1.1 and nwsapi 2.2.2: https://stackblitz.com/edit/node-sahv91?file=package.json&file=index.mjs

<img width="1087" alt="image" src="https://user-images.githubusercontent.com/24715727/236422828-c5c36309-d025-4c85-b20d-659a41c222b0.png">

Online playground with JSDOM 21.1.2 and nwsapi 2.2.4: https://stackblitz.com/edit/node-rqxlwe?file=package.json&file=index.mjs

<img width="1087" alt="image" src="https://user-images.githubusercontent.com/24715727/236422978-789b3aff-cddd-4214-be52-fd1dd90c937e.png">

 

### How does similar code behave in browsers?

Run the code above in browsers should print `1 1 1 1`

**Comment by dormesica** [comment]
Hey
I see this was tagged as `maybe fixed but I'm getting a similar issue on version `24.0.0`.

Is there an expectation for resolution?

**Comment by Manvel** [comment]
> Hey I see this was tagged as `maybe fixed` but I'm getting a similar issue on version `24.0.0`.
> 
> Is there an expectation for resolution?

Yeah, created a ticket in nwsapi repo: https://github.com/dperini/nwsapi/issues/100
Also small effort on fixing: https://github.com/dperini/nwsapi/pull/101 

Hope will be resolved soon, so we can update tooling in our company.

**Comment by asamuzaK** [comment]
jsdom changed the CSS selector engine to [dom-selector](https://github.com/asamuzaK/domSelector) in [Version 23.2.0](https://github.com/jsdom/jsdom/releases/tag/23.2.0), but changed back to nwsapi in [Version 24.0.0](https://github.com/jsdom/jsdom/releases/tag/24.0.0).
So the issue remains.

The easiest workaround is to add `overrides` field in package.json and re-install jsdom.

```json
"overrides": {
  "nwsapi": "2.2.2"
}
````

Or [monkey patch jsdom](https://github.com/asamuzaK/domSelector?tab=readme-ov-file#monkey-patch-jsdom).

**Comment by anlesk** [comment]
I can see that issue still happening.
